Proposed product descriptions/SDK_User_Metrics_and_Analytics.md

```diff
@@ -0,0 +1,52 @@
+## SDK User Metrics & Analytics
+
+### Summary
+KBase needs to have the ability to determine usage of the methods implemented via the SDK.  In the long term, this may include capturing parameters and characteristics of data sets to allow for determination of the quality of results against benchmarks and being able to compare different algorithms and parameterizations to learn which approaches work best.  In this short term, we aim to capture which methods are greatest serving our customer (both in terms of number of runs and unique users), which methods are growing or waning in popularity, whether methods need improvement or are otherwise alienating the customer from their use, and which methods are being used in concert, allowing us to perhaps tailor them to better integrate them as well as train a next-method recommendation feature for the Narrative.  Execution data is captured by the Catalog service, and can now be queried within code cells to produce graphs within Narrative sessions.
+
+### Story/Description
+The initial candidate tables and graphs we propose would include the following:
+
+- Most popular [Methods, Modules] by [Runs, Favorites, Unique-users-per-day]
+  - All Time
+  - This Year
+  - This Month?
```

Comment:
OK, recommendations included.  Thanks!

- Dylan


On Wed, May 11, 2016 at 10:41 AM, Nomi Harris <notifications@github.com>
wrote:

> In Proposed product descriptions/SDK_User_Metrics_and_Analytics.md
> <https://github.com/kbase/roadmap/pull/55#discussion_r62892504>:
>
> > @@ -0,0 +1,52 @@
> > +## SDK User Metrics & Analytics
> > +
> > +### Summary
> > +KBase needs to have the ability to determine usage of the methods implemented via the SDK.  In the long term, this may include capturing parameters and characteristics of data sets to allow for determination of the quality of results against benchmarks and being able to compare different algorithms and parameterizations to learn which approaches work best.  In this short term, we aim to capture which methods are greatest serving our customer (both in terms of number of runs and unique users), which methods are growing or waning in popularity, whether methods need improvement or are otherwise alienating the customer from their use, and which methods are being used in concert, allowing us to perhaps tailor them to better integrate them as well as train a next-method recommendation feature for the Narrative.  Execution data is captured by the Catalog service, and can now be queried within code cells to produce graphs within Narrative sessions.
> > +
> > +### Story/Description
> > +The initial candidate tables and graphs we propose would include the following:
> > +
> > +- Most popular [Methods, Modules] by [Runs, Favorites, Unique-users-per-day]
> > +  - All Time
> > +  - This Year
> > +  - This Month?
>
> Why do the next three have question marks?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kbase/roadmap/pull/55/files/2df0d10a3e8bffe4eae9cf2d782f49bed9e94c2a#r62892504>
>